### ISSUE TITLE
Mismatched parameters in Property definition do not generate an error

### ISSUE THREAD

**Issue by FullValueRider** [WaynePhillipsEA/twinbasic#444]
**Describe the bug**
Using incompatible Types in the definition of a Class Property does not generate an error

**To Reproduce**
```
Public Property Get Code() As ParseResultCode
	Code = p.Code
End Property

Public Property Let Code(ByVal ipCode As ParseResult)
	p.Code = ipCode
End Property
```

In the code above ParseResult is a Class with a predefined id.  PareseResultCode is the name of an Enumeration

**Expected behavior**
The mismatch should be identified as an error

**Desktop (please complete the following information):**
 - OS: [Windows 10]
 - twinBASIC compiler version [v0.10.5788]

**Comment by bclothier** [comment]
I agree this is problematic but that is also valid in VBx so for backward compatibility it has to be supported. This would be managed with `Option Strict` or something like that to avoid mixing different enums or object types without an explicit cast.

**Comment by FullValueRider** [comment]
In VBA you get a compile error.  Different example in VBA but with error messag
![PropertyErrror](https://user-images.githubusercontent.com/81257042/134174633-66c9b447-b5cf-4234-b820-eddb60e60e91.jpg)
e

**Comment by bclothier** [comment]
Sorry, I was hasty in my first reading not realizing this was about property's parameters. Apologies.

**Comment by FullValueRider** [comment]
No worries.  Its useful to add the error message for clarification.
